File: diretorioprincipal/questao14.go
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao14

package main

import (
    "fmt"
    "sync"
)

type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

func GetInstancia() *Configuracao {
    once.Do(func() {
        instancia = &Configuracao{}
        fmt.Println("Configuraçoes carregadas")
    })
    return instancia
}

func (c *Configuracao) ExibirConfiguracao() {
    fmt.Println("mostrando configuraçoes")
}

func main() {
    config1 := GetInstancia()
    config2 := GetInstancia()

    fmt.Println(config1 == config2)  
    config1.ExibirConfiguracao()
}

File: diretorioprincipal/questao14.py
#atividade 06/09/24
#aluno: João Pedro Araújo Inácio
#questao14

class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super(Configuracao, cls).__new__(cls)
            print("configuraçoes carregadas")
        return cls._instancia

    def exibir_configuracao(self):
        print("exibindo as configuraçoes")

config1 = Configuracao()
config2 = Configuracao()

print(config1 == config2) 
config1.exibir_configuracao()
